domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClientStreamResult.java

```diff
@@ -0,0 +1,110 @@
+package org.folio.rest.persist;
+
+import io.vertx.core.Handler;
+import org.folio.rest.jaxrs.model.ResultInfo;
+
+/**
+ * Upon successful completion of PostgresCLient.streamGet, the result is this
+ * class.
+ *
+ * @param <T> each item returned in stream is of this type
+ */
+class PostgresClientStreamResult<T> {
+
+  private final ResultInfo resultInfo;
+
+  private Handler<T> streamHandler;
+  private Handler<Void> endHandler;
+  private Handler<Throwable> exceptionHandler;
+
+  /**
+   * Only to be constructed from PostgresClient itself
+   *
+   * @param resultInfo
+   */
+  PostgresClientStreamResult(ResultInfo resultInfo) {
+    this.resultInfo = resultInfo;
+  }
+
+  /**
+   * The ResultInfo for the streamGet.
+   *
+   * @return Result information.. including totalRecords and facets
+   */
+  public ResultInfo resultInto() {
+    return this.resultInfo;
+  }
+
+  /**
+   * Sets stream handler for each returned item of type T
+   *
+   * @param streamHandler
+   * @return this instance (fluent)
+   */
+  public PostgresClientStreamResult<T> handler(Handler<T> streamHandler) {
```

Comment:
Can you either
* rename `handler` to a more descriptive name, for example `itemHandler`, or
* implement some interface where that name is needed (https://github.com/eclipse-vertx/vertx-sql-client/blob/3.8.5/vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/RowStreamImpl.java#L59 implements `io.vertx.core.streams.ReadStream<T>`)

Reply:
It was modelled after ReadStream<T>, so I'll PostgresClientStreamResult will implement that.